src/py/mat3ra/made/tools/convert/utils.py

```diff
@@ -37,3 +40,30 @@ def extract_tags_from_ase_atoms(atoms: ASEAtoms) -> List[Union[str, int]]:
         int_tags = [int(tag) for tag in atoms.arrays["tags"] if tag is not None]
         result = map_array_to_array_with_id_value(int_tags, remove_none=True)
     return result
+
+
+def _create_pymatgen_structure_from_molecule(ase_atoms: ASEAtoms) -> PymatgenStructure:
+    """
+    Creates a pymatgen Structure from a molecule (non-periodic ASE Atoms).
+
+    Args:
+        ase_atoms (Atoms): An ASE Atoms object with PBC=False.
+
+    Returns:
+        Structure: A pymatgen Structure with CUB lattice sized to contain the molecule, centered.
+    """
+    positions = ase_atoms.get_positions()
+    max_distance = np.max(pdist(positions)) if len(positions) >= 2 else 10.0
+    lattice_size = 2.0 * max_distance
+
+    molecule_center = get_center_of_coordinates(positions.tolist())
+    lattice_center = [lattice_size / 2.0, lattice_size / 2.0, lattice_size / 2.0]
+    translation_vector = np.array(lattice_center) - np.array(molecule_center)
+    centered_positions = positions + translation_vector
+
+    pymatgen_lattice = PymatgenLattice.from_parameters(
+        a=lattice_size, b=lattice_size, c=lattice_size, alpha=90, beta=90, gamma=90
+    )
```

Comment:
The logic used to calculate the padded lattice should be independent of any third-party